libpod/kube.go

```diff
@@ -765,7 +765,17 @@ func simplePodWithV1Containers(ctx context.Context, ctrs []*Container, getServic
 			if !podmanOnly && define.IsReservedAnnotation(k) {
 				continue
 			}
-			kubeAnnotations[fmt.Sprintf("%s/%s", k, removeUnderscores(ctr.Name()))] = v
+			// Certain annotations should be applied to the whole pod.
+			// For others, add container name as a suffix.
+			// For annotations such as this, error if already set.
+			if k == define.UserNsAnnotation {
+				if oldV, ok := kubeAnnotations[k]; ok && oldV != v {
+					return nil, fmt.Errorf("two or more containers have differing user namespace configuration, cannot place in same Kubernetes pod: %w", define.ErrInvalidArg)
+				}
+				kubeAnnotations[k] = v
+			} else {
+				kubeAnnotations[fmt.Sprintf("%s/%s", k, removeUnderscores(ctr.Name()))] = v
+			}
```

Comment:
I'm going to play the grumpy old man yelling "keep off my grass" for a moment.  The code is fine, but I detest one letter variables, especially so if they are used multiple times across more than one or two lines.  I realize `k` and `v` were predefined in the code,  but in instances like this, please consider expanding them to perhaps `key` and `value`, just to make them much easier to pick out of the code, and IMO, makes it an easier read/review.

I now return back to my recliner to get back to napping.